src/strings/utf8.c

```diff
@@ -384,17 +384,18 @@ MVMuint32 MVM_string_utf8_decodestream(MVMThreadContext *tc, MVMDecodeStream *ds
                     break;
                 }
                 case UTF8_REJECT: {
-                    char *waste[] = { (char *)buffer, NULL };
-                    if (bufsize >= 3) {
-                        MVMGrapheme32 a = buffer[pos - 2], b = buffer[pos - 1], c = buffer[pos];
+                    MVMint32 error_pos = pos - 1;
+                    char *waste[] = { (char *)bytes, NULL };
+                    if (error_pos >= 3) {
+                        MVMGrapheme32 a = bytes[error_pos - 2], b = bytes[error_pos - 1], c = bytes[error_pos];
```

Comment:
Why MVMGrapheme32? char and MVMGrapheme32 are both signed, so converting the value will sign extend, i.e. every char with the high bit set will be printed as e.g. ffffff80.